datastore/get_metas.go

```diff
@@ -0,0 +1,159 @@
+package datastore
+
+import (
+	"github.com/PretendoNetwork/nex-go"
+	common_globals "github.com/PretendoNetwork/nex-protocols-common-go/globals"
+	datastore "github.com/PretendoNetwork/nex-protocols-go/datastore"
+	datastore_types "github.com/PretendoNetwork/nex-protocols-go/datastore/types"
+)
+
+func getMetas(err error, client *nex.Client, callID uint32, dataIDs []uint64, param *datastore_types.DataStoreGetMetaParam) uint32 {
+	if commonDataStoreProtocol.getObjectInfoByDataIDHandler == nil {
+		common_globals.Logger.Warning("GetObjectInfoByDataID not defined")
+		return nex.Errors.Core.NotImplemented
+	}
+
+	if commonDataStoreProtocol.verifyObjectPermissionHandler == nil {
+		common_globals.Logger.Warning("VerifyObjectPermission not defined")
+		return nex.Errors.Core.NotImplemented
+	}
+
+	if err != nil {
+		common_globals.Logger.Error(err.Error())
+		return nex.Errors.DataStore.Unknown
+	}
+
+	// TODO - Verify if param.PersistenceTarget is respected? It wouldn't make sense here but who knows
+
+	pMetaInfo := make([]*datastore_types.DataStoreMetaInfo, 0, len(dataIDs))
+	pResults := make([]*nex.Result, 0, len(dataIDs))
+
+	// * param has an AccessPassword, but it goes unchecked here.
+	// * The password would need to be the same for every object
+	// * in the input array, which doesn't make any sense. Assuming
+	// * it's unused until proven otherwise
+
+	for i := 0; i < len(dataIDs); i++ {
+		objectInfo, errCode := commonDataStoreProtocol.getObjectInfoByDataIDHandler(dataIDs[i])
+
+		if errCode != 0 {
+			// TODO - Maybe this should be broken out into a util function in globals?
```

Comment:
I agree, I think we could make a helper that creates an empty `DataStoreMetaInfo`

Reply:
I was thinking about this just now, it would take a lot of work to go through all the types but this should honestly probably be implemented in `nex-protocols-go`. Right now all the types `New` functions return zeroed structs, which works fine for primative types like numbers and strings. Those will default to their zero. But in structs which have fields that are pointers to other structs that's when this breaks down since that results in a nil pointer

The `New` functions should honestly be updated to just return properly zeroed structs rather than ones containing nil pointers. That way we can just call the `New` function to get our empty structs

Reply:
I agree, that would be the best solution

Reply:
I added this to all the `New` functions in stock DataStore and SMM (since that's what this is aimed at right now). The others can come later as we need them, it's not relevant to add zeroed struct support to every protocol right now